src/Integration/Binding/ProjectSystemFilter.cs

```diff
@@ -0,0 +1,106 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ProjectSystemFilter.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using DteProject = EnvDTE.Project;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    internal class ProjectSystemFilter : IProjectSystemFilter
+    {
+        private readonly IProjectSystemHelper projectSystem;
+
+        private Regex testRegex;
+
+        public ProjectSystemFilter(IHost host)
```

Comment:
From a UX and troubleshooting point of view, it would be useful to send info about excluded/test projects to the output window (otherwise the user is liable to wonder why some projects weren't bound).